R/pmcmc.R

```diff
@@ -696,3 +699,165 @@ plot.pmcmc <- function(x, ...) {
   
   
 }
+
+##' @export
+##' @importFrom viridis cividis
+##' @importFrom graphics hist par plot.new text lines legend
+##' 
+plot.pmcmc_list <- function(x, burn_in = 1, ...) {
+  
+  summ <- summary(x, burn_in = burn_in)
+  par_names <- names(x$inputs$pars$pars_init)
+  n_pars <- length(par_names)
+  
+  chains <- x$chains
+  n_chains <- length(chains)
+  cols_trace <- viridis::viridis(n_chains)[c(4, 1, 3, 2)]
+  
+  
+  # compile master chain and order by log posterior for plotting
+  master_chain <- do.call(what = rbind, 
+                          args = lapply(X = chains, 
+                                        FUN = function(z) {
+                                          z$results[-seq_len(burn_in), ]
+                                        }
```

Comment:
Put all these closing brackets together, or indent the last one in a clearer place